springcloud/src/main/java/org/axonframework/extensions/springcloud/commandhandling/mode/IgnoreListingDiscoveryMode.java

```diff
@@ -101,7 +120,21 @@ public void updateLocalCapabilities(ServiceInstance localInstance,
     }
 
     private boolean shouldIgnore(ServiceInstance service) {
-        return ignoredServices.stream().anyMatch(ignoredService -> equals(ignoredService, service));
+        long currentTime = clock.instant().toEpochMilli();
+        boolean shouldIgnore = false;
+        Set<ServiceInstance> expiredInstances = new HashSet<>();
+
+        for (Map.Entry<ServiceInstance, Long> instance : ignoredServices.entrySet()) {
+            if (equals(instance.getKey(), service)) {
```

Comment:
Nit: Can be moved out of the loop and replaced with `containsKey`

Reply:
I'm going to leave this one as is, as the `equals` method that's invoked is a custom one from the `IgnoreListingDiscoveryMode`.
Although I think there's a way to make it so the `containsKey` method would invoke that exact `equals` operation, I need to do the loop for entry removal anyhow.
So, although I agree, I think this fine for now.